### PR TITLE
feat: Add silence alerts feature

### PR DESCRIPTION
**Pull request by teeohhem**
Fixes: HDX-2734

### CODE REVIEW

File: packages/api/src/tasks/checkAlerts/index.ts [hunk 1]
```diff
@@ -101,11 +101,16 @@ const fireChannelEvent = async ({
     throw new Error('Team not found');
   }
 
-  if ((alert.silenced?.until?.getTime() ?? 0) > Date.now()) {
+  // Re-fetch the alert to get the latest silenced state. Other alert properties
+  // (threshold, interval, groupBy, etc.) are stable during execution and do not
+  // need to be refetched. We only check the silence state here to ensure we respect
+  // user silencing actions that may have occurred after the alert task was queued.
+  const freshAlert = await Alert.findById(alert.id);
```
Comment (pulpdrew) [inline packages/api/src/tasks/checkAlerts/index.ts]: As per [this issue](https://linear.app/clickhouse/issue/HDX-2281/alert-further-decouple-mongo-from-the-alert-check-loop-oss-repo), we have removed all direct mongo/docdb references from the alerting code and are using the alert provider to fetch data from mongo.

It's possible that original motivation for that ticket ("We're going to eventually have to avoid having alert job pods connecting back to mongo/documentdb.") is no longer true, in which case this is fine. @dhable do you know if this is still a requirement?

File: packages/app/src/api.ts [hunk 2]
```diff
@@ -92,6 +92,23 @@ const api = {
         }),
     });
   },
+  useSilenceAlert() {
+    return useMutation<any, Error, { alertId: string; mutedUntil: string }>({
```
Comment (pulpdrew) [inline packages/app/src/api.ts]: Would it make sense for this to be a Date?
```suggestion
    return useMutation<any, Error, { alertId: string; mutedUntil: Date }>({
```

File: packages/app/src/AlertsPage.tsx [hunk 3]
```diff
@@ -73,6 +88,175 @@ function AlertHistoryCard({
...
+
+  if (alert.silenced?.at) {
+    return (
+      <ErrorBoundary message="Failed to load alert acknowledgment menu">
+        <Menu>
+          <Menu.Target>
+            <Button
+              size="compact-sm"
+              variant="light"
+              color={isNoLongerMuted ? 'orange' : 'green'}
```
Comment (pulpdrew) [inline packages/app/src/AlertsPage.tsx]: Should we use the shared color variables here (and elsewhere above?)
```suggestion
              color={
                isNoLongerMuted
                  ? 'var(--color-bg-warning)'
                  : 'var(--color-bg-success)'
              }
```

File: packages/app/src/AlertsPage.tsx [hunk 4]
```diff
@@ -73,6 +88,175 @@ function AlertHistoryCard({
...
+              leftSection={<i className="bi bi-bell-slash fs-8" />}
```
Comment (pulpdrew) [inline packages/app/src/AlertsPage.tsx]: Should we use the Tabler icon here? (And `IconBellOff` for the `bell-slash` below)?
```suggestion
              leftSection={<IconBell size={16} />}
```

File: packages/app/src/DBDashboardPage.tsx [hunk 5]
```diff
@@ -247,7 +259,7 @@ const Tile = forwardRef(
                     color="gray"
                     size="xxs"
                     onClick={onEditClick}
-                    title="Alerts"
+                    title={alertTooltip}
```
Comment (pulpdrew) [inline packages/app/src/DBDashboardPage.tsx]: Optionally, it might be nice to convert this to a mantine `Tooltip` around the button instead of using the button title - I find the titles always require hovering too long to show up.

File: packages/app/src/DBDashboardPage.tsx [hunk 6]
```diff
@@ -207,6 +207,17 @@ const Tile = forwardRef(
       return 'red';
     }, [alert]);
 
+    const alertTooltip = useMemo(() => {
+      if (!alert) {
+        return 'Add alert';
+      }
+      let tooltip = `Has alert and is in ${alert.state} state`;
+      if (alert.silenced?.by) {
+        tooltip += `. Ack'd by ${alert.silenced.by}`;
```
Comment (pulpdrew) [inline packages/app/src/DBDashboardPage.tsx]: This seems to just be a user ID, which is not very user-friendly. Can we populate the user name/email or if not, should we just say use the date, like `Ack'd at ${alert.silenced.at}`?

Screenshot 2025-12-12 at 2 57 10 PM

Comment (teeohhem) [reply packages/app/src/DBDashboardPage.tsx]: Good idea. Let me just update it to `Ack'd at date` for simplicity

File: packages/app/src/DBDashboardPage.tsx (hunk 5 above)
Comment (teeohhem) [reply packages/app/src/DBDashboardPage.tsx]: Good idea. Will change

File: packages/api/src/tasks/checkAlerts/index.ts (hunk 1 above)
Comment (teeohhem) [reply packages/api/src/tasks/checkAlerts/index.ts]: Good call. I remember this now. I'll update

Comment (teeohhem) [reply packages/api/src/tasks/checkAlerts/index.ts]: removed the db call and commented about the edge case + updated tests

**Review by pulpdrew — APPROVED** [review]
